operator/api/vectorized/v1alpha1/cluster_types.go

```diff
@@ -196,6 +202,43 @@ type ClusterSpec struct {
 	NodePools []NodePoolSpec `json:"nodePools,omitempty"`
 }
 
+// ClusterConfiguration holds values (or references to values) that should be used
+// to configure the cluster. Where the cluster schema defines a non-string type for a
+// given key, the corresponding values here should be string-encoded (according to yaml
+// rules)
+type ClusterConfiguration map[string]ClusterConfigCRDValue
+
+type YAMLRepresentation string
+
+// ClusterConfigCRDValue represents a value of arbitrary type T. Values are string-encoded according to
+// YAML rules in order to preserve numerical fidelity.
+// Because these values must be embedded in a `.bootstrap.yaml` file - during the processing of
+// which, the AdminAPI's schema is unavailable - we endeavour to use yaml-compatible representations
+// throughout. The octet sequence of a Representation will be inserted into a bootstrap template
+// verbatim.
+// TODO: this type should be lifted to a shared module rather than duplicated in the `redpanda` CRD definition.
+type ClusterConfigCRDValue struct {
```

Comment:
nit: Feels funky to include CRD in the name of a value that's included as part of a CRD. _Hopefully_ the package name and/or some kubebuilder annotations are enough to showcase that this is part of a CRD.
```suggestion
type ClusterConfigValue struct {
```

Reply:
OOC what is the expected behavior if all provided members are `nil`? If that's an invalid case could you add a TODO to add in CEL validation thereof?

Reply:
Done